src/domains/WinningLotto.js

```diff
@@ -0,0 +1,92 @@
+const Messages = require('../constants/Messages');
+const LottoError = require('../errors/LottoError');
+const Lotto = require('./Lotto');
+const Reward = require('./Reward');
+
+class WinningLotto {
+  /** @type {Lotto} */
+  #lotto;
+
+  /** @type {number} */
+  #bonusNumber;
+
+  /** @type {Reward[]} */
+  #availableRewards;
+
+  /**
+   * @param {Lotto} lotto
+   * @param {number} bonusNumber
+   * @param {Reward[]} availableRewards
+   */
+  constructor(lotto, bonusNumber, availableRewards = Reward.DEFAULT_REWARDS) {
+    this.#lotto = lotto;
+    this.#bonusNumber = bonusNumber;
+    this.#availableRewards = availableRewards;
+    this.validate();
+  }
+
+  validate() {
+    this.#validateBonusNumberType();
+    this.#validateBonusNumberDuplication();
+  }
+
+  /**
+   * @param {Lotto} lotto
+   * @returns {number}
+   */
+  countMatchNumber(lotto) {
+    const [numbers, otherNumbers] = [this.#lotto.getNumbers(), lotto.getNumbers()];
+    const count = numbers
+      .map((number) => (otherNumbers.includes(number) ? 1 : 0))
+      .reduce((a, b) => a + b, 0);
```

Comment:
사소한 부분이지만 위의 sort 함수에서도 `(a,b) => a - b` 형식으로 작성하셨는데 변수 명을 조금 더 의미있는 것으로 바꾸었다면 좋았을 것 같다는 생각이 들어요!

Reply:
```suggestion
      .reduce((numberA, numberB) => numberA + numberB, 0);
```
확실히 a, b만으론 의미를 알기 어려울 것 같아요..! number 이라는 수식어를 붙여봤는데 a, b는 대체할게 없는거같네요...ㅎㅎ

Reply:
다른 분들 코드를 보니 `front, back`이나 `next, prev` 등으로 선언하셨더라구요!

Reply:
그런 이름도 있군요! 감사합니다~

Reply:
이건 @solo5star @Leemainsw 두 분의 의견이 궁금한 부분인데요! 

<img width="151" alt="image" src="https://user-images.githubusercontent.com/48716298/202840480-9e5f3a48-066a-4345-ad02-2cd87a2d852f.png">

> 출처: https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce

reduce에서 받는 네 가지 인수들은 accumulator, currentValue, currentIndex(optional),array (optional) 이에요. 

용혁님께서 a, b 라고 선언한 변수들은 실제로는 accumulator(누산기), currentValue(현재 값)라는거죠! 
근데 `a`, `front`, `prev` 라는 네이밍으로 누산기라는 의미가 충분히 전해질 수 있을지 의문이에요...

reduce가 **기존 값에 계속 더해 결과를 도출하는 함수**라고 생각이 들어 `acc`라는 네이밍을 살리는 것이 좋을 것 같다는 개인적인 의견이 드는데 두 분은 어떻게 생각하시나요?

+) 개인적으로 sort에서 a, b는 이미 [sort의 설명 네이밍](https://developer.mozilla.org/ko/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#%EC%84%A4%EB%AA%85)이 a, b 이기도 하고 a와 b는 동일 선상에 있다는 느낌이 들어 사용하기 좋아보입니다!

Reply:
@areumsheep 님 의견 감사합니다! 말씀하신 대로 `원소 자체의 의미`보다는 `동일한 두 원소` 를 표현하는데 a, b가 적합하다고 생각하여 사용했습니다. (물론 원소의 의미는 알 수 없습니다) 이를 reduce에서도 동일하게 생각하여 a, b를 사용했는데 reduce에서는 a가 사실 동일한 원소는 아니고 누산된 값이기 때문에 부적절할 수도 있을 것 같습니다. 바꾼다면 널리 사용되는 `acc`, `cur` 정도가 어떨까... 싶습니다.

Reply:
의견 공유해주셔서 감사드립니다. 저도 acc, cur에 동의합니다!